Client/Ecosia/Analytics/Analytics.swift

```diff
@@ -29,134 +29,46 @@ final class Analytics {
         })
     }
-    
-    func screen(_ label: Label.Screen) {
-        tracker.track(SPStructured.build {
-            $0.setCategory(Category.navigation.rawValue)
-            $0.setAction(Action.view.rawValue)
-            $0.setLabel(label.rawValue)
-        })
-    }
-    
-    func newTab(_ origin: Property) {
-        tracker.track(SPStructured.build {
-            $0.setCategory(Category.browser.rawValue)
-            $0.setAction(Action.open.rawValue)
-            $0.setLabel("new_tab")
-            $0.setProperty(origin.rawValue)
-        })
-    }
-    
-    func features(_ label: Label.Features, origin: Property) {
+
+    func browser(_ action: Action.Browser, label: Label.Browser, property: Property? = nil) {
         tracker.track(SPStructured.build {
             $0.setCategory(Category.browser.rawValue)
-            $0.setAction(Action.open.rawValue)
-            $0.setLabel(label.rawValue)
-            $0.setProperty(origin.rawValue)
-        })
-    }
-    
-    func market(_ new: String) {
-        tracker.track(SPStructured.build {
-            $0.setCategory(Category.navigation.rawValue)
-            $0.setAction("change")
-            $0.setLabel("market")
-            $0.setProperty(new)
-        })
-    }
-    
-    func more(_ label: Label.More) {
-        tracker.track(SPStructured.build {
-            $0.setCategory(Category.navigation.rawValue)
-            $0.setAction(Action.open.rawValue)
+            $0.setAction(action.rawValue)
             $0.setLabel(label.rawValue)
         })
+
     }
-    
-    func shareApp(_ action: Action.Share) {
+
+    func navigation(_ action: Action, label: Label.Navigation) {
         tracker.track(SPStructured.build {
             $0.setCategory(Category.navigation.rawValue)
             $0.setAction(action.rawValue)
-            $0.setLabel("share_app")
-        })
-    }
-    
-    func shareContent(_ action: Action.Share) {
-        tracker.track(SPStructured.build {
-            $0.setCategory(Category.browser.rawValue)
-            $0.setAction(action.rawValue)
-            $0.setLabel("share_content")
+            $0.setLabel(label.rawValue)
         })
     }
 
-    func favourites(_ action: Action.Favourites) {
-        tracker.track(SPStructured.build {
-            $0.setCategory(Category.browser.rawValue)
-            $0.setAction(action.rawValue)
-            $0.setLabel("favourites")
-        })
-    }
-    
-    func history(_ action: Action.History) {
-        tracker.track(SPStructured.build {
-            $0.setCategory(Category.browser.rawValue)
-            $0.setAction(action.rawValue)
-            $0.setLabel("history")
-        })
-    }
-    
-    func tabs(_ action: Action.Tabs) {
-        tracker.track(SPStructured.build {
-            $0.setCategory(Category.browser.rawValue)
-            $0.setAction(action.rawValue)
-            $0.setLabel("tabs")
-        })
-    }
-    
-    func news(_ id: String) {
+    func navigationOpenNews(_ id: String) {
```

Comment:
I would rename this method. Maybe is not necessary to add the navigation part to it. Does it help to know it is part of 'navigation' when you are trying to open a news item? I would name it `openNews(_ id: String)`

Reply:
My intention was to reflect the (main) category of the events (browser, navigation, external, migration) also in the prefix of the  method namings. That way correlations are easier to find when comparing the documentation and the code. When having convenient methods I would aim at reflecting all the params in the method name. So this name here reflects category: navigation, action: open, label:news. If you don't mind too much, I'd like to try it that way to ease up future maintenance.

Reply:
I see it as an implementation detail, and I think longer names not necessarily means easing maintenance. But I have no strong objections about using these long names so we can keep them 👍